### ISSUE TITLE
CostMap obstacle gradient

### ISSUE THREAD

**Issue by warenick** [warenick/DDP_learning#3]
I added static map data as [CostmapReader](https://github.com/warenick/DDP_learning/blob/master/scripts/utilites/CostmapReader.py) class object to Agents objects for calculation the loss function: ([Agent.py](https://github.com/warenick/DDP_learning/blob/master/scripts/env/Agent.py))
```
    def running_cost(self, state, controll, others=None, k_state=1):
        state_cost = torch.clip(self.final_cost(state)*k_state, 0, 10)
        controll_cost = torch.clip(torch.sum(torch.pow(controll, 2)), 0, 10)
        social_cost = torch.clip(self.social_cost(state, others) if others is not None else torch.tensor(0.), 0, 10)
        costmap_cost = self.costmap_cost(state, controll) if self.costmap is not None else torch.tensor(0.)
        return state_cost+controll_cost+social_cost+controll_cost*costmap_cost
```

costmap_cost calculated just by taking current position cost:

```
    def costmap_cost(self, state, controll):
        return torch.tensor(self.costmap.at_position(state))
```
That's become just a constant value in DDP gradients calculations and because of that, I multiply costmap cost on control cost for the non-zero gradient.
Overall gradient works, but its direction is invariably backward across agent moving, but not in direction of the obstacle gradient map. In that case, cosmap cost helps only for a slow downing agent nearby collisions, but not for avoidance or detouring the obstacles.

How I can change gradient direction or something else to detour the obstacles? 

https://user-images.githubusercontent.com/7687321/148955164-63378a63-a68b-484b-a7f6-e57b5b6e2d22.mp4
